src/amberscript/parser.cc

```diff
@@ -989,5 +993,119 @@ Result Parser::ParseClear() {
   return ValidateEndOfStatement("CLEAR command");
 }
 
+Result Parser::ParseExpect() {
+  auto token = tokenizer_->NextToken();
+  if (token->IsEOS() || token->IsEOL())
+    return Result("missing pipeline name in EXPECT command");
+  if (!token->IsString())
+    return Result("invalid pipeline name in EXPECT command");
+
+  auto* pipeline = script_->GetPipeline(token->AsString());
+  if (!pipeline)
+    return Result("unknown pipeline name for EXPECT command");
+
+  token = tokenizer_->NextToken();
+  if (token->IsEOS() || token->IsEOL())
+    return Result("missing BUFFER in EXPECT command");
+  if (!token->IsString() || token->AsString() != "BUFFER") {
+    return Result("expected BUFFER got '" + token->ToOriginalString() +
+                  "' in " + "EXPECT command");
+  }
+
+  token = tokenizer_->NextToken();
+  if (!token->IsString())
+    return Result("invalid buffer name in EXPECT command");
+
+  auto* buffer = script_->GetBuffer(token->AsString());
+  if (!buffer)
+    return Result("unknown buffer name for EXPECT command");
+
+  token = tokenizer_->NextToken();
+  if (!token->IsString() || token->AsString() != "IDX")
+    return Result("missing IDX in EXPECT command");
+
+  token = tokenizer_->NextToken();
+  if (!token->IsInteger() || token->AsInt32() < 0)
+    return Result("invalid X value in EXPECT command");
+  uint32_t x = token->AsUint32();
+
+  token = tokenizer_->NextToken();
+  if (!token->IsInteger() || token->AsInt32() < 0)
+    return Result("invalid Y value in EXPECT command");
+  uint32_t y = token->AsUint32();
+
+  // TODO(dsinclair): Handle comparator && TOLERANCE
+
+  token = tokenizer_->NextToken();
+  if (token->IsString() && token->AsString() == "SIZE") {
+    bool found = false;
+    for (const auto& info : pipeline->GetColorAttachments()) {
+      if (info.buffer == buffer) {
+        found = true;
+        break;
+      }
+    }
+    if (!found)
+      return Result("buffer not in pipeline for EXPECT command");
+
+    auto probe = MakeUnique<ProbeCommand>(pipeline, buffer);
+    probe->SetX(x);
```

Comment:
MSVC is complaining about possible loss of data on the conversion from uint32_t to float.
Here and later on.